yi-keymap-vim/src/Yi/Keymap/Vim/Ex/Commands/BufferCycle.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Yi.Keymap.Vim.Ex.Commands.BufferCycle (parse) where
+
+import           Control.Applicative                 (Alternative ((<|>)))
+import           Control.Monad
+import           Data.List
+import qualified Data.Attoparsec.Text             as P
+import           Lens.Micro.Platform
+import           Yi.Editor
+import           Yi.Keymap                        (Action (EditorA))
+import           Yi.Window
+import           Yi.Keymap.Vim.Common                (EventString)
+import qualified Yi.Keymap.Vim.Ex.Commands.Common as Common
+import           Yi.Keymap.Vim.Ex.Types           (ExCommand (cmdAction, cmdShow))
+
+parse :: EventString -> Maybe ExCommand
+parse = Common.parse $ do
+  void (P.char 'b') <|> return ()
+  direction <- (fmap (const False) $ P.string "prev" <|> P.string "previous") <|> fmap (const True) (P.string "next")
```

Comment:
`P.string "prev" <|> P.string "previous"` is redundant, because the "prev" parser also matches "previous". Maybe this behavior should be changed, because now "prevsomething" is also accepted.

Reply:
This can be solved by using the [`endOfInput`](http://hackage.haskell.org/package/attoparsec-0.13.2.2/docs/Data-Attoparsec-ByteString-Char8.html#v:endOfInput) parser

Reply:
I opened https://github.com/yi-editor/yi/issues/1085 to track this issue for all parsers.